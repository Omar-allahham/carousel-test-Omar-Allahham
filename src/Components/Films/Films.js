
import React, { useEffect, useState } from "react";
import Carousel from "../Carousel/Carousel";
import ErrorMessage from "../ErrorMessage";

export default (props) => {
    const [state, setState] = useState({ films: [], isLoading: true, APIError: false, ErrorMessage: "" });
    useEffect(() => {
        fetch(`https://swapi.dev/api/films/`, {
            method: "GET"
        })
            .then((res) => res.json())
            .then((response) => {
                setState({ films: response.results, isLoading: false });
            })
            .catch((error) => setState({ ...state, APIError: true, ErrorMessage: error.message }));
    }, []);

    if (state.APIError === true)
        return <ErrorMessage message={state.ErrorMessage} />

    return (<Carousel items={state.films} isLoading={state.isLoading} ItemTemplate={FilmCarouselItem} />);
};

function FilmCarouselItem(props) {
    const { item, index } = props;

    return (
        <div className={props.classes} >
            <h1>{item.title}</h1>
            <h2><a href={`/characters/${index}`}> Characters</a></h2>
        </div>
    );
}







