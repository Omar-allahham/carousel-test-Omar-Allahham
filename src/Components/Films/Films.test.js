import React from 'react';
import renderer from 'react-test-renderer';
import Films from './Films';


describe('Films Component', () => {
    it('match the snapshots', () => {
        const tree = renderer
            .create(<Films />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

})



