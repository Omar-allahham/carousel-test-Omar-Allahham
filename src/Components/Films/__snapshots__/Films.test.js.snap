// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films Component match the snapshots 1`] = `
<div
  className="carousel-0-2-1"
>
  <div
    className="navButton-0-2-2"
    onClick={[Function]}
  >
    <i
      className="arrowLeft-0-2-3"
    />
  </div>
  <div
    className="ldsRipple-0-2-10"
  >
    <div />
    <div />
  </div>
  <div
    className="navButton-0-2-2"
    onClick={[Function]}
  >
    <i
      className="arrowRight-0-2-4"
    />
  </div>
</div>
`;
